### PR TITLE
Add a loginWithTLS function to V1

### PR DESCRIPTION
**Pull request by souleb**
Adresses [#217](https://github.com/oras-project/oras/issues/217)

If implemented, a client will accept custom certificate and certificate authority to authenticate with a remote registry.

We ended up copying several unexported functions from upstream in order to be able to implement this.

Signed-off-by: Soule BA <bah.soule@gmail.com>

### CODE REVIEW

File: pkg/auth/docker/login.go [hunk 1]
```diff
@@ -78,9 +80,21 @@ func (c *Client) login(settings *iface.LoginSettings) error {
...
+		_, token, err = c.loginWithTLS(ctx, remote, settings.CertFile, settings.KeyFile, settings.CAFile, &cred, userAgent)
+		if err != nil {
+			return err
+		}
+	} else {
+		_, token, err = remote.Auth(ctx, &cred, userAgent)
+		if err != nil {
+			return err
+		}
+	}
```
Comment (shizhMSFT) [inline pkg/auth/docker/login.go:95]: nit:
```suggestion
	if (settings.CertFile != "" && settings.KeyFile != "") || settings.CAFile != "" {
		_, token, err = c.loginWithTLS(ctx, remote, settings.CertFile, settings.KeyFile, settings.CAFile, &cred, userAgent)
	} else {
		_, token, err = remote.Auth(ctx, &cred, userAgent)
	}
	if err != nil {
		return err
	}
```

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Comment by sajayantony** [comment]
@souleb @shizhMSFT any concerns merging this?

**Comment by souleb** [comment]
> @souleb @shizhMSFT any concerns merging this? 

@shizhMSFT requested @deitch review before merging.

**Pull request merged**